engine/univGen.ml

```diff
@@ -130,7 +130,7 @@ let fresh_inductive_instance env sp =
 let fresh_constructor_instance env sp =
   fresh_constructor_instance env sp None
 
-let constr_of_global gr =
+let constr_of_monomorphic_global gr =
   let c, ctx = fresh_global_instance (Global.env ()) gr in
     if not (Univ.ContextSet.is_empty ctx) then
       if Univ.LSet.is_empty (Univ.ContextSet.levels ctx) then
```

Comment:
Should we keep accepting polymorphic values with empty instances or should we start testing with is_polymorphic?

Reply:
Fixed